chainer/training/triggers/interval.py

```diff
@@ -40,9 +42,16 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch_detail = updater.epoch_detail
+            previous_epoch_detail = updater.previous_epoch_detail
```

Comment:
> Nodes in the abstract computational graph correspond to chainer.Variables and operators to chainer.Functions.

I understand some nodes are not called for each iteration. However, from my understanding, `Variable` and `Function` do not have any `Trainer` objects. How do they call a trigger internally?